connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/importer/ProcessDefinitionInspector.java

```diff
@@ -158,8 +163,17 @@ private Optional<ProcessCorrelationPoint> getCorrelationPointForElement(
   }
 
   private Optional<ProcessCorrelationPoint> getCorrelationPointForIntermediateCatchEvent(
-      IntermediateCatchEvent catchEvent) {
+      IntermediateCatchEvent intermediateCatchEvent) {
+    return getCorrelationPointCatchEventEvent(intermediateCatchEvent);
+  }
+
+  private Optional<ProcessCorrelationPoint> getCorrelationPointForIntermediateBoundaryEvent(
+      BoundaryEvent boundaryEvent) {
+    return getCorrelationPointCatchEventEvent(boundaryEvent);
+  }
```

Comment:
Can we remove the wrappers and call `getCorrelationPointCatchEvent` directly here?

Reply:
Good point. I am thinking about it and just trying to wrap my head around multiple message start events too. Because start event is technically catch event as well, so we might want to reuse that code too. Just exploring this flow tho.

Reply:
If you don't mind, I will address this in the https://github.com/camunda/team-connectors/issues/468, as that one requires some refactoring too.